src/store/types.d.ts

```diff
@@ -129,24 +132,20 @@ export interface ActiveMediaState {
   status: 'ejected' | 'playing' | 'paused' // 'ejected' means player is closed
 }
 
-export interface MediaStoreResult {
-  count: number
-  page?: number
-  pageCount: number
-  items: AudioDetail[] | ImageDetail[]
-}
+export interface MediaStoreResult<T extends FrontendMediaType>
+  extends MediaResult<Record<MediaDetail['id'], T>> {}
 
 export interface MediaState {
   results: {
-    audio: MediaStoreResult
-    image: MediaStoreResult
+    audio: MediaStoreResult<AudioDetail>
+    image: MediaStoreResult<ImageDetail>
```

Comment:
These need to refer to a `FrontendMediaType` value.

```suggestion
    audio: MediaStoreResult<'audio'>
    image: MediaStoreResult<'image'>
```